### DISCUSSION TITLE
👨‍💻☎ WinUI Community Call: June 21, 2023

### DISCUSSION THREAD

**Discussion by gabbybilka** [microsoft/microsoft-ui-xaml#8487] [Announcements]
### Please join us for the June 2023 edition of the WinUI Community Call. This event will be live-streamed to YouTube: https://www.youtube.com/live/4CzUfD1YL8A

# Details
Date: June 21, 2023
Time: 16:00-17:00 UTC (9:00-10:00am Pacific)

Anyone and everyone is welcome - no pre-registration is required.
This will be an informal interactive live stream directly with members of our engineering team.

# Format
The community call is a call among the WinUI team that is live-streamed onto YouTube. We present on new updates, share information, welcome guests, and answer your questions.

### **Agenda**
- Intro
- WinAppSDK releases & updates
- Q&A

### **Q&A Code of Conduct**
- To help ensure Q&A remains beneficial for the majority of the call's developer audience, please consider whether your question could be framed so that it might be more broadly relevant to our larger developer audience by ensuring it is reasonably concise, broadly beneficial, well-suited for the fast-paced live Q&A format, or able to spark open ended discussion.  
- Please understand some questions are not well-suited for the fast-paced live Q&A format and should be filed as standalone [Q&A](https://github.com/microsoft/microsoft-ui-xaml/discussions/new?category=q-a&title=%5BYour+Question+Here%5D) discussions where they can be most appropriately handled. Excluding presentation topic requests, this generally includes requests for dedicated follow-ups or technically complex questions that require investigative time or expertise not included in this month's guest list.

### Leave us your questions in the comments on this issue or live in the YouTube chat during the stream!

**Comment by lukeblevins** [comment]
When will the roadmap be properly updated?

**Reply by TRadigk** [reply]
The roadmap is shown by gabby in the beginning of every community call. You can check the latest video for the latest roadmap. Usually later on upcoming improvements are also mentioned. If you talk about the product board page: it is deprecated, which was topic in more than one community call.

**Reply by llothar** [reply]
Thats not a roadmap. But this is really the one used internally by Microsoft, it would explain the disaster that WinUI3 and WinAppSdk is at the moment.

**Reply by riverar** [reply]
The WinUI team suggested that roadmap and open source news would be made available at Microsoft Build 2023. I missed [the discussion](https://build.microsoft.com/en-US/sessions/6d3467f6-a6a9-42b9-a404-c9ed507b6066?source=sessions) and, of course, the discussion wasn't recorded. I heard from attendees, however, that those items were not shared.

So... we're back to the original question.

**Comment by Move2win** [comment]
Face your most commented issue - Add WinUI 3.0 XAML Designer in Visual Studio.

**Reply by roxk** [reply]
Even in UWP I never used it. IMO they should put the resource elsewhere, like improve hot reload.

**Reply by Move2win** [reply]
> Even in UWP I never used it. IMO they should put the resource elsewhere, like improve hot reload.

IMO, a designer makes life easier, especially when lining up/alignment UI elements, you can drag them and preview instantly without touching any codes. That makes all the difference.
Also, older frameworks like WinForm and WPF have designers, but the newer WinUI3 doesn't. I see that as a downgrade/regression.

**Reply by jamespack** [reply]
I agree with @roxk. I'd much rather see the resources it would take for a designer be spent elsewhere.

**Reply by Move2win** [reply]
> I agree with @roxk. I'd much rather see the resources it would take for a designer be spent elsewhere.

@jamespack Development efficiency or toolchains is much much much more important than some random feature. The team should take more care of the tools first, then other features. If you don't have a good devtool, how are people gonna get in? Don't your guys understand that?

**Reply by jamespack** [reply]
I've never actually considered the designer a good dev tool. The one in blend was ok I guess but I always found myself dropping down the xaml anyway. The rest of the tool chain is pretty good in my humble opinion. But there are quite a few straight up missing features or broken features or features that are there but need to be used in a way that is not sufficiently documented. Also, I would presume that the designer is outside of this teams scope.

**Reply by TRadigk** [reply]
This topic was discussed more than once already (if I remember correctly with answer from Bob Pulliam and Ryan Demopoulos) and the designer is not necessarily an issue of Winui/WASK but with Visual Studio Dev Team. Since the community (we) reported a low use of the designer for UWP (any maybe even WPF) this is not a priority. I personally think it is pretty readable *if* developers use speaking names for (dependency) properties.
Without putting too much effort into the search, I think this might be the latest answer: https://youtu.be/xcgc5CLqUkM?t=3092

**Reply by F8RDev** [reply]
> This topic was discussed more than once already (if I remember correctly with answer from Bob Pulliam and Ryan Demopoulos) and the designer is not necessarily an issue of Winui/WASK but with Visual Studio Dev Team. Since the community (we) reported a low use of the designer for UWP (any maybe even WPF) this is not a priority. I personally think it is pretty readable _if_ developers use speaking names for (dependency) properties. Without putting too much effort into the search, I think this might be the latest answer: https://youtu.be/xcgc5CLqUkM?t=3092

I found it hard to believe WPF developer is not using designer, most WPF application is UI based, if not then they just use Console App.
When creating a complex  WPF app with many views, like a business app will find a designer very handy.

Maybe the UWP developer is mostly from Web Dev?
So they already live without or didn't care about visual design?

I think the problem is not whether this feature is important or not, but this feature is not an MS priority.
And from what I heard, the team is quite small, so I have less confidence in WinApp/WinUI than when I jump into UWP platform.

**Reply by llothar** [reply]
>  Since the community (we) reported a low use of the designer for UWP (any maybe even WPF) this is not a priority.  

So you produce a shitty tool and then use the refusal of people using it as an indicator that it isn't important? 
And this was on a system that failed itself and never used for the target audience that they now want to reach (the reunion with the majority of business apps).  

Yeah sounds like great modern software management. Why not check Xcode and Android Studio to see how many people there would say, no we don't use it, plain code is good enough for me.

**Reply by pjmlp** [reply]
> Even in UWP I never used it. IMO they should put the resource elsewhere, like improve hot reload.

You mean like the C++/WinRT tooling promised to us during the CppCon 2016, as means to make up for deprecating C++/CX?

Or the XAML islands and XAML direct from 2019?

Better wait seated.

**Reply by nikolayvpavlov** [reply]
There are rumors on Reddit that MAUI will be scrapped to give way to Blazor Native.  Scrapping MAUI is a good thing.  It is a mistake that should have never happened.  Xaml world was in a sad state already and there was no need to introduce another dialect, with another platform, when moving on with UWP would have been the better idea.  But, the direction that Office and Teams is taking - web apps, be it Blazor or React or whatever, is hideous.  I am yet to see a web app performing better than a native one, and on shared environments like Azure Virtual Desktop or Citrix, web-apps just bring the servers down.

**Comment by nikolayvpavlov** [comment]
We are dearly missing:
1. Stable, high-performant data-grid control, as available in both WinForms and WPF.
2. Modern ribbon toolbar, like Office apps.
3. Inking control.

Are there plans to facilitate interoperability with Win32?  WinForms and WPF both have native classes and APIs, which makes interop with Win32 easier.  We need an easier way to catch Windows messages, without subclassing the main window using Win32 APIs.  Overriding a method like WndProc will be good enough.  More examples: WinForms has an ImageList class, which makes it easier to integrate with the Shell and Taskbar.  In WASDK we need to go all the way down to bare-metal Win32.

**Reply by roxk** [reply]
Just curious, is the data grid in the community toolkit not working for you?

**Reply by nikolayvpavlov** [reply]
@roxk, the grid in WCT is ok, though I have some concerns about its performance.  Still, it is a community effort, for which the roadmap and support are kind of unclear.  We need to develop LoB apps that we must support, and I am not willing to rely on something that may not get fixes / updates at all.

**Reply by roxk** [reply]
WCT is from MS, by MS.

https://learn.microsoft.com/en-us/dotnet/communitytoolkit/introduction

> The toolkit is maintained and published by Microsoft, and part of the .NET Foundation

**Reply by llothar** [reply]
> Just curious, is the data grid in the community toolkit not working for you?

It is not working for anyone using C++ with the WinUI3 toolkit that is written in C++. Or Rust or any other language. 

Don't want to add a non portable monster runtime when the major revenue comes in from MacOS and Windows is just a port. Yes i said it now, windows is just 2nd ... for a few years already.  MS has not the power anymore to enforce their tools and languages technologies and on the world. Tides have changed.

**Reply by llothar** [reply]
I second the need for a working inking toolkit. And offline device local text recognition software.

**Reply by pjmlp** [reply]
@llothar You're better off with Qt or if budget allows, VCL/Firemonkey.

**Reply by nikolayvpavlov** [reply]
I wouldn't touch VLC/Firemonkey with a three-meter pole.  The last trial version I tried crashed on the splash screen. :-)

**Reply by omikhailov** [reply]
> Just curious, is the data grid in the community toolkit not working for you?

It is of low quality, but that's not even the point. DataGrid is one of the key components, like ListView, TextBox or Button, and it is at least strange that it is not part of the standard set of controls available out of the box. Let's look at it from the other side: why not move the Button into Community Toolkit then? Because it would be silly, so why should we reference some library to use DataGrid?

**Comment by TRadigk** [comment]
I would like to ask if the team could make it possible to post release notes in the style the maui team does. There a mention is done to address fixed issues. Lately I've found several fixed issues, that were resolved silently. On the official release notes it then looks like "mhm, not much has happened, I think I can skip this release", but unknowingly something was changed (for the better).

Example: https://github.com/dotnet/maui/releases/tag/8.0.0-preview.3.8149

**Comment by eduardobragaxz** [comment]
I'm sorry if this is out of scope, but could anyone give us an update on WebAuthenticationBroker? Is it still being worked on for WinUI 3 apps?

**Comment by r7dev** [comment]
I want to hear news about Input Validation support. #179
Will we have a happy ending?

**Reply by knightmeister** [reply]
This is the never ending saga, such a crucial function for LOB applications but it's ignored because it's not shiny

**Reply by r7dev** [reply]
I don't understand what's going on in their minds. LOB applications spin trillions of dollars worldwide. 💵🌎

**Reply by nikolayvpavlov** [reply]
Obviously they have drank too much of hipster's kool-aid that web apps are all the world needs and wants...  It is tragedy.

**Reply by r7dev** [reply]
Just as airplane was designed to fly and ship was designed to sail on the high seas. Web and Desktop applications have also been designed for your different demands. This new generation of Redmonds are like ants without antennae. 🐜📡❌

**Reply by dongle-the-gadget** [reply]
I thought they have already supported it considering that C#/WinRT maps INotifyDataErrorInfo to WinUI 3.

**Comment by sotanakamura** [comment]
Please add XamlApplication API, an application implementing IXamlMetadataProvider interface. It allows us to use WinUI 3 controls easily from a code with any language projections and no XAML. #8475

**Comment by riverar** [comment]
[The Fluent 2 session](https://build.microsoft.com/en-US/sessions/71f477ad-2fe1-4907-b387-f589a0b582a1?source=sessions) at Build did not mention how to use Fluent 2 in WinUI. Is this available in WinUI 2 or 3 releases today? Or scheduled for later?

**Reply by ghost1372** [reply]
did you see https://fluent2.microsoft.design/ ?

**Reply by riverar** [reply]
Yeah, I saw the 13 words on the whole site. Doesn't answer my question.

**Reply by ghost1372** [reply]
Hmm...
I can see that fluent 2 is for winui 3

I think Fluent 2 is designed more for the web part, in some of their documentation (for windows) they have linked to the WinUI 3 documentation.

**Reply by riverar** [reply]
@ghost1372 The images are too big, can you delete them? I'm asking for precise targeting information--what version of WinUI/WinAppSDK is needed to take advantage of Fluent 2. Will it be supported on Windows 10? Are all controls updated? etc. etc. etc.

**Reply by ghost1372** [reply]
> @ghost1372 The images are too big, can you delete them? I'm asking for precise targeting information--what version of WinUI/WinAppSDK is needed to take advantage of Fluent 2.

Mybe version 1.4-experimental1
They released this version in nuget a few days ago

**Reply by riverar** [reply]
@ghost1372 Thanks, will check it out!

**Reply by mdtauk** [reply]
I am not sure if Fluent 2 offers anything specific to WinUI - but that WinUI 3 is being included under the "Fluent 2" family.

Fluent 2 seems mostly to be the React and Web UI platform and could be used for PWA web apps that look consistent and good on Windows 11.

AFAIK there are no changes coming to existing WinUI control templates.

**Comment by gus33000** [comment]
Is the information available on https://microsoft.github.io/microsoft-ui-xaml up to date?

For example in the about page it is mentioned WinUI 3 has support for pen input, but i'm not finding much documentation online on InkCanvas availability in winui 3.
![image](https://github.com/microsoft/microsoft-ui-xaml/assets/3755345/50a6b155-6441-40de-8497-6ec3a896d99a)

I'm also noticing outdated information like the date of the next community call in the main page, and the discord link on the page. It seems that the winui team is not as active as it used to be on this discord channel. Is this something still being looked at?

![image](https://github.com/microsoft/microsoft-ui-xaml/assets/3755345/3b8a1d2b-9b14-4a0e-9e24-30b0b297c8ce)

**Reply by gus33000** [reply]
The about page also mentions webview2 is not available for winui 2 but this is not the case currently anymore

**Comment by JanRajnoha** [comment]
April 12, day when Windows App SDK version 1.3 has been released. That day, I reported problem with touch keyboard, which still not show for PasswordBox. @gabbybilka asked @codendone and @Scottj1s for ability to repro problematic behavior, but they ignored her. Two other people had same problem and we waited for your answer, but nothing happened. I tried to ask again @gabbybilka, but I have been ignored. 
1. Are you listening developers? 
2. When we could see some resources dedicated to fix this bug
3. Are you planning something to prevent this situations? Because this happen too often.

Issue: https://github.com/microsoft/microsoft-ui-xaml/issues/6291

**Comment by Mysterious-Dev** [comment]
With recent changes for servicing update (1.3.1) for title in XAML, can we imagine a better way to set the minimum size of an application with an XAML version (like we can do with WinUiEx) for 1.4 or 1.5 ?

**Comment by nikolayvpavlov** [comment]
The latest Insider builds of Windows 11 feature the new File Explorer with Xaml islands and WinUI 3 content.  Would you please report on the status of Xaml Islands / Content and when and how these can be used in Win Forms and WPF applications?  What are the restrictions?  Will third-party controls like WCT work within Islands?

**Reply by eduardobragaxz** [reply]
The WinAppSDK version of File Explorer doesn't use Xaml islands. It doesn't need to.

edit: ok I checked. It seems it uses both WinUI 2 and 3 at the same time? I don't know, it's weird.

**Reply by nikolayvpavlov** [reply]
> edit: ok I checked. It seems it uses both WinUI 2 and 3 at the same time? I don't know, it's weird.

The only way to achieve this is to use some kind of "island".  If we can do this, too, embed WinUI 3 controls / presenter into a Win32 (read Windows Forms or WPF) app, it would be great.  Right now, complete WinUI 3 apps are a pain to develop.  Too much interop, too much going back to the old ages of Win32... Worse than MFC.

**Reply by llothar** [reply]
Is File Explorer open source already?  I did not get the impression in the last Community Call that they use Xaml or WinUI2.

**Reply by nikolayvpavlov** [reply]
No, and it never will be, if you ask me.  It is too tightly integrated with a core OS component: the Shell.

**Reply by dongle-the-gadget** [reply]
File Explorer uses WinUI 3, whereas other bits of explorer.exe (like the Taskbar) use WinUI 2.

This is made possible by an internal API within Windows.

**Comment by mdtauk** [comment]
Any more updates on a good (if not best in class) design time experience for WinUI app design

**Reply by riverar** [reply]
We received an update at the last standup or so. It's a large task and there are no plans to work on it. It's not happening. Folks should instead get comfy with XAML hot reload.

**Reply by mdtauk** [reply]
> We received an update at the last standup or so. It's a large task and there are no plans to work on it. It's not happening. Folks should instead get comfy with XAML hot reload.

I understand that, but will continue to ask until I get an answer I am happy with lol

**Reply by jamespack** [reply]
> We received an update at the last standup or so. It's a large task and there are no plans to work on it. It's not happening. Folks should instead get comfy with XAML hot reload.

Good. I'd rather see that time spent else where. 😁

**Comment by MEK3DK** [comment]
Can you comment on the rumors comming out of Build about the future of WinUI 3? That WinUI 3 future might be unsure since MAUI seems to be adding a WPF backend?

**Reply by nikolayvpavlov** [reply]
@MEK3DK , what is this about MAUI adding a WPF backend?

**Reply by MEK3DK** [reply]
I did not attend Build but somebody posted it on Twitter, I kind of ignored that. Now somebody also posted it in this forum, so I would like the team to just give a quick answer. Probably the same rumor just going around. In some way it makes sense to add WPF, but also a large and unlikely task.

**Comment by riverar** [comment]
RichTextLinks have been broken in touch scenarios for 2 years. What's the status of https://github.com/microsoft/microsoft-ui-xaml/issues/6513?

**Reply by gautambjain** [reply]
The fixed this in latest WinAppSDK 1.3.2 announced today - https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/stable-channel#version-132-13230602002

**Reply by riverar** [reply]
Thanks for the heads up. There's still the lingering question about servicing UWP/WinUI 2. Hope that gets answered, good or bad.

**Comment by knightmeister** [comment]
This is probably going to be contentious but it feels to me as someone who has followed WinUI since it was "open sourced" - and before the product had a name - that progress has pretty much stopped. No updated roadmap in ages. Critical (for LOB apps) things like validation still not done. There are issues I've raised over 4 years ago which are still open and unaddressed.

Looking at the open issues is worrying; very rarely does the team comment and the majority of issues that have been opened aren't responded to, and when they are they are usually by non-MSFT people.

Is there any plans to address the growing backlog of issues that have been raised?
Can we get some clarity around roadmap and plans for the future?
When will the source for WinUI 3 be made available so at least the community can do work where Microsoft is unable/unwilling to do so?

**Comment by gautambjain** [comment]
I have read hundreds of posts over last few months related to disappointment caused due to WinUI 3's non-maintenance, non-response, careless attitude. Not having money and resources for this shows their low-level of confidence on WinUI 3. Or probably AI and Cloud computing is just too lucrative that they have become insensitive to Windows developers.

Again and again I have my doubts that Microsoft may just buy out another UI framework and drop WinUI 3?

**Reply by ChaimCL** [reply]
Abandoning the AOT compilation technology is something I still find hard to understand (maybe related technology will be added later), UWP has many flaws, but it is perfectly possible to shift its direction to focus on the desktop to improve it, instead of building another wheel. The current winui3 features are extremely immature and lacking in startup performance, and I'm having a hard time being convinced that this is what the developers want.I'm not so sure they'll acquire another UI framework (but that's a good thing, as it would unintentionally make things worse), and obviously the focus is now on cloud & AI.

**Reply by llothar** [reply]
You are right, desktop isn't on their mind anymore, but they will nevertheless happy to make double digit  billions from 100 Euro/License fees for Windows 12. The whole last Build Conference was total boring for someone who has a business that relies on user data which is not uploaded to the cloud. Not even an amazing Visual Studio talk anymore. 

The time of Steve Ballmers Developer monkey dance are long over. Can't believe that i say, i miss crazy Steve.

**Reply by Panda-Sharp** [reply]
Steve Ballmer made some mistake, but we cannot not say that he was passionate and really cared about windows (and windows development)

**Reply by eduardobragaxz** [reply]
I don't know how it works with C++, but better AOT support is something that's being worked on for C# and (hopefully) consequently C#/WinRT.

**Reply by nikolayvpavlov** [reply]
@llothar , there is nothing wrong with having the data in the cloud, but businesses still need a snappy, powerful apps to work with.  Web, advanced as it is, still offers inferior UX, esp. when it comes to integration with Office and particularly Excel.  Now, in my app, I can let users just select a range in the grid and copy & paste it into Excel.  Good luck doing so in a web app without asking for user permissions, or downloading a file.  And streaming clipboard content from a web-app?  Please.

**Reply by DarranRowe** [reply]
@eduardobragaxz 
AOT is ahead of time compiled. Basically, this means that instead of being compiled to il and then having the native code generated at a later point, it is instead compiled to the native binary directly.
C++ works like this by default. The output of the compiler/linker process is a native binary.

**Reply by eduardobragaxz** [reply]
@DarranRowe thanks!

**Comment by LoriBru** [comment]
How do you expect people to work with and be excited about WinUI if even Microsoft itself is going in the opposite direction and abandoning native apps and replacing them with web apps?
https://support.microsoft.com/en-us/office/outlook-for-windows-the-future-of-mail-calendar-and-people-on-windows-11-715fc27c-e0f4-4652-9174-47faa751b199

**Reply by nikolayvpavlov** [reply]
This is so depressing that I cry.  

And, no, I don't buy the excuse that Office is a huge, separate team, a company within the company.  Same goes for Teams: again a slow, memory-guzzler, a web app, the new one, too.  There must be a company-wide strategy for applications.  For Christ sake, Teams doesn't even implement Fluent UI.  

P.S.  I tried the new Outlook.  It is slower than the native version, consumes more memory, and lacks simple features like copy & paste a file from Windows Explorer into a new message as attachment.  I cringed and hastily returned back to the old Outlook.  

P.P.S. And it is touch-unfriendly.  I tried it in tablet mode, and I cringed even more.

**Reply by Panda-Sharp** [reply]
I can forgive only Teams to be web based, it's huge, complex, lot integrations and cross platform... But for all the rest of apps there are no excuses, it's really sad, since whiteboard moved from native to web became unusable

**Reply by riverar** [reply]
Opinion: Outlook is unifying their code base across all platforms to make it easier to update, fix bugs, add value, etc. The common UI stack across all supported platforms is web, not WinUI. That said, there's an argument to potentially use something else that spits out native integrations for each platform, but I don't think there's anything mature enough to rely upon in that area.

**Reply by nikolayvpavlov** [reply]
@riverar , opinion:  I agree with yours, but now we see sacrifice of stability, features, UX and design over faster to the market. For a key product of a multi-billion dollar company that is just pathetic.  MS thrived while supporting Office for Mac and Windows independently, and all of a sudden they go to the web.  Why? To support 1% of Linux users who would not use Office anyway?  

Investing in WinUI 3 and supporting platforms like Uno would make much more sense.  Client Blazor relies a lot on WASM and there is no reason not to follow Uno's approach and make WinUI 3 available on the web via WASM.

**Reply by llothar** [reply]
Faster to market? Why? Is there any competition out there for Outlook? 

It's just the a very sad state (for us technical minded people) but end users dont seem to care anymore. The only complaint if any is that you have to use the cloud. And comparing   WinUI or Web will not make  a difference when coming from Win32. Everything is now a monster  memory and performance wise.

**Comment by ackh** [comment]
What is the status of open sourcing the Windows App SDK? Seriously, there needs to be an answer to this question. Every month that passes without it being open source is another month of community effort wasted.

[Rumors are popping up](https://www.reddit.com/r/dotnet/comments/14dnqlh/winui3_is_dead_windev_mvp_calling_for_it_to_be/) that WinUI / WinAppSdk is dead. I doubt that this is true but if you don't properly communicate what is going on and what happens when, this is what you get.

**Reply by riverar** [reply]
Tagging @dotMorten who can maybe speak to his comments directly.

**Reply by dotMorten** [reply]
I was being a bit silly in that post (as the tweet said I meant to write something else but let auto complete have some fun). It’s not dead. What I mean to say is we’re not seeing the resources on WinUI that it deserves. It appears heavily understaffed. Just look at [the web Authenticator issue](https://github.com/microsoft/WindowsAppSDK/issues/441) - It’s a non stop list of comments from people asking for an update to the issue, and it's severely impacting .NET MAUI.
I’m just frustrated to see something that has so great bones only get small bug fixes and only the features added are for 1st party apps like file explorer or a few key partner apps. I feels as if there is as good as zero resources dedicated to fixing what the community is asking for over and over again, and the Sdk is becoming a long list of disappointing broken promises.
I still believe winui3 could be great - but the way it is going it won’t ever be without a change in the goals and vision of the project. 
its frustrating to see all these resources poured into making Windows great for developers like DevDrive, terminal, devhome etc but no one stops to think what message the WinAppSDK is sending to the same developers

**Reply by llothar** [reply]
The Visual Studio team just revealed their engineering size in a blog post. 

They have 26 Software Developer and 6 Produkt Managers and one technical writer in their Team that is only working on the Editor Component of Visual Studio.

And WinAppSDK barely fixed the hyperlink click bug in the RichEditText after reported/confirmed  two years ago.

Please reveal your team size.

**Reply by llothar** [reply]
"its frustrating to see all these resources poured into making Windows great for developers like DevDrive, terminal, devhome etc but no one stops to think what message the WinAppSDK is sending to the same developers" 

Come on, if you aren't blind you see the business idea behind that. They need all the source code  they can get their fingers on to train an AI that does all the development work (on their proprietary servers) and sell this lot of money on API calls.  There are  alternatives for  Visual Studio but there wont be for the AI because our governments are bribed or blind and sleeping on the AI train.

But  there is absolute no business model for a GUI toolkit. Why you think that Qt was sold for less then 30 million dollar? 
People will upgrade to windows 12 no matter if a new toolkit is there or not.  WinUI3 would adds zero additional business opportunities for Microsoft. With WinUI2 at least they hoped for billions from their own Appstore and a 30% mafia tax on all sales of other peoples hard work.

 Sad is that even Qt is not an option, because not only does it  look bad and is now only focused on embedded devices, for better Ink Controls, Touch and Fluid Design we need the core OS/Driver integration to make this experience smooth.
 
 I hope one day the government understands and splits Microsoft and Amazon, maybe all of the FAANGs.

**Reply by omikhailov** [reply]
>But there is absolute no business model for a GUI toolkit.

By this logic, Asp.Net and .Net are also not needed, but Microsoft is investing huge resources in them. Frameworks are needed as a key part of the ecosystem, which is sold as a whole, in one package and not in one product. The only problem is that today all of .Net is dedicated to the web, but this team sits in the corner and does not participate in the party.

>I hope one day the government understands and splits Microsoft and Amazon, maybe all of the FAANGs.

That would be good for IT industry, but bad for America, so it won't happen soon.

**Comment by ackh** [comment]
I have accepted the fact that the there is no XAML designer and that there won't be one for several years. I have also accepted the fact that the C++ tooling to write WinAppSdk apps is severely lacking. But what really gets to me is that bugs that we open are just going nowhere.

The SwapChainPanel is broken since WinAppSdk 1.2 as described [here](https://github.com/microsoft/microsoft-ui-xaml/issues/7924). I have been waiting for a fix for this for more than 6 months. While some effort was made, there are still problems. Unfortunately no one seems to care. I'm still stuck on WinAppSdk 1.1.5 due to that bug which is why I really need someone to look into this and respond.

**Reply by riverar** [reply]
Opinion: Microsoft maintains public and private repositories for WinUI, WinAppSdk, etc. The public repository, on GitHub, is not critical to the team's workflow nor their primary work surface so it's naturally prone to neglect. I suspect there are a number of other issues compounding matters, such as not having time/resources to address any external bugs, all hands on deck for the File Explorer _customer_, sinking morale due to key employee departures + unhappy customers (us), and upper management pressure to deliver something that smells like AI.

To Microsoft: You still have an open offer from me to help clean up the repository.

**Comment by llothar** [comment]
> To Microsoft: You still have an open offer from me to help clean up the repository.

Yeah, i would also help (for a salery and if 100% online work is accepted) with bug fixing?  Are you hiring Microsoft ?

**Comment by Noemata** [comment]
From the rumblings I've heard from insiders, WinUI is effectively dead.  The bigger problem for Microsoft these days is the xOS juggernaut (macOS, iOS, tvOS, watchOS, visionOS, etc.).  Apple has so outpaced Microsoft, Windows itself has become a losing proposition and will collapse as a viable product within a few years.

**Reply by Poopooracoocoo** [reply]
Desktop Windows isn't going anywhere. I'd say ChromeOS/Android is a bigger threat, or maybe iPadOS. Native Windows development is a different story and the web is the future of Windows apps, however bleak it is. Microsoft seems to be okay with Windows being on the back burner and likely doesn't want to compete with Apple

**Reply by elankath** [reply]
It is utterly strange - Apple is investing all-in in native apps. SwiftUI gets major updates every half-year and more and more devs are shifting to macOS. They then build more apps for iOS/watchOS/etc. They are pulling market share from Microsoft and Microsoft remains un-concerned. Are they no longer a OS company ? Do they only care about cloud subscription revenue now ? But that is a losing game in the long term as after the base rots way, there is nothing to distinguish them from the hundreds of cloud companies out there.

**Comment by pubtom** [comment]
When will be the next WinUI Community Call?
Can you provide a github link?

**Reply by JanRajnoha** [reply]
@chigy maybe could give some info

**Reply by chigy** [reply]
@pubtom , @JanRajnoha , we are planning on next call on 9/27. We will start a new community call thread soon.

**Comment by Noemata** [comment]
Here's my last attempt at trying to salvage Windows as a viable platform.  UWP was the correct path forward.  It only needed five fixes.

1) Make general file system access an automatic opt out option for all UWP apps.  The user must take an action through settings to revoke general file system access for any given UWP app.  This would have resolved a significant usability pain point for UWP apps.

2) Add a full trust capability that allows for complete system wide access outside the UWP sandbox.  This is essentially what WinUI is.  There was absolutely no need to re-invent the wheel and destroy UWP's viability in the process.  Most apps only really needed general file system access, so a full trust capability could be enabled on a case by case basis as apps are deployed to the Microsoft store.  Given sideloading is already an open feature on Windows, folks that wanted to bypass an approval process still can.

3) Resolve the remaining well known edge cases of the .Net native compiler so 99% of UWP apps could be stable.

4) Bring back UWP to the Windows system level UI for OS services, settings, file manager, etc.  The Windows 11 (back to Windows 7) idea will fail miserably.

5) Stop the WinUI, MAUI nonsense.  Consolidate all development around UWP.  UWP was the right path forward.  Win32 and the days when apps could be trusted are long gone.

Lastly, WinUI and MAUI are both a disaster.  I for one will not look at Windows again until UWP is brought back.  Which probably means never given how little Microsoft pays attention to its most loyal developers.  This is coming from a long time Windows fanboy, loyalist and advocate.  ALL my development now revolves around Apple tech.

The most recent release of SwiftUI has brought it to parity with the things I liked about UWP.  What was once a significant lead for UWP has been reduced to parity.  Microsoft still has the best IDE (Visual Studio), and the most efficient programming language (C#), but the repeated lapses toward self-destruction within the organization makes those looking outside in think the company has a death wish.

Unfortunately for the Windows/Intel/PC Vendors eco system, Apple's vertical and horizontal integration is turning into a Mount Everest of tech.  Microsoft is not solely to blame for Windows demise.  Enterprise customers have had too much financial sway and influence over the company.  Intel, because they had Apple on their silicon, stayed too neutral and stopped investing in Windows.  Next year, PC sales will collapse if there isn't a major shift by Microsoft/Intel.  It will be a very quick end to the Windows PC business.

Accept that mistakes were made.  Bring back UWP, and stop the madness.  Also, stop listening to your enterprise customers.  They do not innovate, they merely follow trends.

Oh, and if Microsoft thinks their cloud initiatives will save the day, with Apple in control of the entire consumer space, they will eventually wipe out all of Microsoft's enterprise business as they too switch away from Microsoft/Intel.  AI, unless it can start creating hardware/software that is superior to what Apple has on offer will be too little, too late.

**Reply by llothar** [reply]
Apple is nice and some businesses (creative agencies and tech) do use them but Apple has even much less respect for their customers and their product management will never fit into the large parts corporate or government workspace. Also they offer zero choice of customization, which makes it impossible. They are in their business actions till pure home use/lifestyle.

Nadella is a cloud guy and now cloud will get even stronger. No AI models from MS will ever run locally in custom apps (unless we will have a giant step forward). And they with "npm" and "github" are already at cartel like levels of monopolization. 

There is no money reason for MS to change their current successfull path.

**Reply by nikolayvpavlov** [reply]
> There is no money reason for MS to change their current successfull path.

On the contrary.  The MS strategy is indeed to move everything to cloud and web, but native apps are still a big market and alienating developers from their platform is also a good trigger to make them go to Amazon and Google, and not to Azure.  One of Azure's strengths is the integration with on-premises MS technologies and easy migration.  Have no people develop for Windows, and you can find yourself that no one wants to use Azure SQL Database.

**Reply by omikhailov** [reply]
> There is no money reason for MS to change their current successfull path.

Nobody said they should close the cloud business.

**Comment by pubtom** [comment]
Always hard to find the next one:
WinUI Community Call: September 27, 2023 https://github.com/microsoft/microsoft-ui-xaml/discussions/8884
